src/main/java/dev/jbang/util/UnpackUtil.java

```diff
@@ -86,21 +87,25 @@ public static void unzip(Path zip, Path outputDir, boolean stripRootFolder, Path
 				if (!entry.startsWith(outputDir)) {
 					throw new IOException("Entry is outside of the target dir: " + zipEntry.getName());
 				}
-				if (zipEntry.isDirectory()) {
-					Files.createDirectories(entry);
-				} else if (zipEntry.isUnixSymlink()) {
-					Scanner s = new Scanner(zipFile.getInputStream(zipEntry)).useDelimiter("\\A");
-					String result = s.hasNext() ? s.next() : "";
-					Files.createSymbolicLink(entry, Paths.get(result));
-				} else {
-					if (!Files.isDirectory(entry.getParent())) {
-						Files.createDirectories(entry.getParent());
-					}
-					if (Files.isRegularFile(entry)) {
-						onExisting.handle(zipFile, zipEntry, entry);
+				try {
+					if (zipEntry.isDirectory()) {
+						Files.createDirectories(entry);
+					} else if (zipEntry.isUnixSymlink()) {
+						Scanner s = new Scanner(zipFile.getInputStream(zipEntry)).useDelimiter("\\A");
+						String result = s.hasNext() ? s.next() : "";
+						Files.createSymbolicLink(entry, Paths.get(result));
 					} else {
-						defaultZipEntryCopy(zipFile, zipEntry, entry);
+						if (!Files.isDirectory(entry.getParent())) {
+							Files.createDirectories(entry.getParent());
```

Comment:
wont this fail the parent exist but is a file?

Reply:
> wont this fail the parent exist but is a file?
> ... wondering if it still wont fail for multi level nested dirs.
> ... but that dir can't be created as its parent-parent is a file

That's why the PR lets the error occur and catches the `FileAlreadyExistsException` instead of trying to test preventively all possible situations where this could happen.

Reply:
as you can see with the failure that is not what happens in this case :)